src/Game.java

```diff
@@ -43,25 +28,36 @@ public void printCharArray() {
     }
 
 
+    public void handleCommand(String input, Hero hero, Stone stone) {
+
+        //check for special words (maybe should store all special words in some place?)
+        if(input.toLowerCase().equals("quit") ||
+                input.toLowerCase().equals("reset")) {
+            return;
+        }
 
-    public void handleCommand(String input) {
         char command = input.charAt(0);
-        Coordinates futureMove = hero.checkRoad(command);
+        if ( command != 'w' && command != 's' && command != 'a' && command != 'd'  ){
+            System.out.println("Command not recognized.");
+            System.out.println("Use W, S, A, D to move around.");
+            System.out.println("Type QUIT to quit the game or RESET to reset the level.");
+            return;
+        }
+
+        Coordinates futureMove = hero.checkRoad(command, hero.position);
 
         //stone
         if (map[futureMove.x][futureMove.y] == 'O') {
-            Coordinates futureStoneMove = stone.checkRoad(command);
+            Coordinates futureStoneMove = stone.checkRoad(command, stone.position);
```

Comment:
Twoim nast celem powinno być wyłuskanie tej logiki gdzieś gdzie będziesz mógł przetestować czy odpowiednie rzeczy się tutaj dzieją, to jest dla Ciebie największy kawałek logiki biznesowej